icu4j/main/classes/core/src/com/ibm/icu/impl/number/LongNameHandler.java

```diff
@@ -61,14 +67,241 @@ static String getWithPlural(String[] strings, StandardPlural plural) {
         return result;
     }
 
+    private enum PlaceholderPosition { NONE, BEGINNING, MIDDLE, END }
+
+    private static class ExtractCorePatternResult {
+        String coreUnit;
+        PlaceholderPosition placeholderPosition;
+        char joinerChar;
+    }
+
+    /**
+     * Returns three outputs extracted from pattern.
+     *
+     * @param coreUnit is extracted as per Extract(...) in the spec:
+     *   https://unicode.org/reports/tr35/tr35-general.html#compound-units
+     * @param PlaceholderPosition indicates where in the string the placeholder
+     *   was found.
+     * @param joinerChar Iff the placeholder was at the beginning or end,
```

Comment:
do you mean `If` or `If and only if`

Reply:
I mean "if and only if".

Reply:
(This doesn't end up in [ICU4J API documentation](https://unicode-org.github.io/icu-docs/apidoc/released/icu4j/).)